### PR TITLE
[dev] vscode debug launch.json

### PR DESCRIPTION
**Pull request by esigo**
Adds vscode launch.json
launch.json is needed for debugging in VS code.

## Changes
how to use:
build in debug mode e.g. bazel:
`--compilation_mode=dbg`

or CMake with:
`-DCMAKE_BUILD_TYPE=Debug`

change the target to the executable of interest e.g. `curl_http_test`:
`"target": "${workspaceFolder}/bazel-bin/ext/test/http/curl_http_test"`

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by owent** [comment]
Should we commit IDE setting to repository?Different users use different IDE and toolchains.

**Comment by esigo** [comment]
> Should we commit IDE setting to repository?Different users use different IDE and toolchains.

We discussed this PR in the SIG meeting. I think it is good to at least have something minimal if everybody agrees of course.

**Comment by marcalff** [comment]
> > Should we commit IDE setting to repository?Different users use different IDE and toolchains.
> 
> We discussed this PR in the SIG meeting. I think it is good to at least have something minimal if everybody agrees of course.

LGTM.

I don't see how this can hurt, and it will increase our productivity.

**Review by marcalff — APPROVED** [review]

**Review by marcalff — APPROVED** [review]

**Comment by reyang** [comment]
> Should we commit IDE setting to repository?Different users use different IDE and toolchains.

Looks like OpenTelemetry C++ is not the first one https://github.com/open-telemetry/opentelemetry-specification/blob/main/.vscode/settings.json.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+        {
+            "name": "Debug",
+            "type": "gdb",
+            "request": "launch",
+            "target": "${workspaceFolder}/bazel-bin/<path to the bin file>",
+            "cwd": "${workspaceRoot}",
+            "valuesFormatting": "parseText"
+        }
+    ]
+}
```
Comment (reyang) [inline .vscode/launch.json:13]: nit: consider adding a blank new line here

**Review by reyang — APPROVED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (esigo) [reply .vscode/launch.json]: thanks, done

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Debug",
+            "type": "gdb",
+            "request": "launch",
+            "target": "${workspaceFolder}/bazel-bin/<path to the bin file>",
```
Comment (ThomsonTan) [inline .vscode/launch.json:8]: This is bazel specific and some some manual update for CMake?

**Review by ThomsonTan — APPROVED** [review]

File: .vscode/launch.json (hunk 2 above)
Comment (esigo) [reply .vscode/launch.json:8]: it can be used with CMake too, just need to change the target path.

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+            "cwd": "${workspaceRoot}",
+            "valuesFormatting": "parseText"
+        }
+    ]
```
Comment (lalitb) [inline .vscode/launch.json:12]: Should we add the configuration for Windows too? Something like this should work.
```
        {
            "name": "Debug on Windows",
            "type": "cppvsdbg",
            "request": "launch",
            "program": "${workspaceFolder}/build/<path-to-bin-file>",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false
        }
```

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Debug",
```
Comment (lalitb) [inline .vscode/launch.json:5]:
```suggestion
            "name": "Debug on Linux",
```

File: .vscode/launch.json (hunk 3 above)
Comment (esigo) [reply .vscode/launch.json:23]: thanks, added

**Pull request merged**